python/ray/metrics_agent.py

```diff
@@ -5,188 +5,110 @@
+
+    def _record_metrics(self, metrics):
+        # The list of view data is what we are going to use for the
+        # final export to exporter.
+        view_data_changed: List[ViewData] = []
+
+        # Walk the protobufs and convert them to ViewData
+        for metric in metrics:
+            descriptor = metric.metric_descriptor
+            timeseries = metric.timeseries
+
+            if len(timeseries) == 0:
+                continue
+
+            columns = [label_key.key for label_key in descriptor.label_keys]
+            start_time = timeseries[0].start_timestamp.seconds
+
+            # Create the view and view_data
+            measure = measure_module.BaseMeasure(
+                descriptor.name, descriptor.description, descriptor.unit)
+            view = self.view_manager.measure_to_view_map.get_view(
+                descriptor.name, None)
+            if not view:
+                view = View(
+                    descriptor.name,
+                    descriptor.description,
+                    columns,
+                    measure,
+                    aggregation=None)
+                self.view_manager.measure_to_view_map.register_view(
+                    view, start_time)
+            view_data = (self.view_manager.measure_to_view_map.
+                         _measure_to_view_data_list_map[measure.name][-1])
+            view_data_changed.append(view_data)
+
+            # Create the aggregation and fill it in the our stats
+            for series in timeseries:
+                tag_vals = tuple([val.value for val in series.label_values])
+                for point in series.points:
+                    if point.HasField("int64_value"):
+                        data = CountAggregationData(point.int64_value)
+                    elif point.HasField("double_value"):
+                        data = LastValueAggregationData(
+                            ValueDouble, point.double_value)
+                    elif point.HasField("distribution_value"):
+                        # print(f"got histogram {descriptor}")
+                        dist_value = point.distribution_value
+                        counts_per_bucket = [
+                            bucket.count for bucket in dist_value.buckets
+                        ]
+                        bucket_bounds = (
+                            dist_value.bucket_options.explicit.bounds)
+                        data = DistributionAggregationData(
+                            dist_value.sum / dist_value.count,
+                            dist_value.count,
+                            dist_value.sum_of_squared_deviation,
+                            counts_per_bucket, bucket_bounds)
+                    else:
+                        raise ValueError("Summary is not supported")
+
+                    view_data.tag_value_aggregation_data_map[tag_vals] = data
+
+        # Finally, export all the values
+        self.view_manager.measure_to_view_map.export(view_data_changed)
```

Comment:
So, are you basically implement this in our way instead of using `record`? Is it because we need to aggregate manually? In this case, I feel like we need more testing for correctness. I think it is a good idea to use `alive_actor` metrics to test the correctness of this logic.

Reply:
I meant when I looked at the documentation (the only thing I could find is this https://opencensus.io/quickstart/python/metrics/#1), it uses measure_map.record(). Also, the opencencus repo said it measure_to_view_map.export is deprecated (https://github.com/census-instrumentation/opencensus-python/blob/e1d3c3b7d86ea8601f59836d0a4ec4c55117d33f/opencensus/stats/measure_to_view_map.py#L128). So, I assume it is not the public API? I might miss something, but in that case, I thought we have to add some tests to make sure it works correctly.

Reply:
And about tests, what I meant was to track alive actors stats and make sure it is properly set. For example, you can create 2~3 actors, check metrics, and kill 1 of them, check it again, or something like that.